Assets/MixedRealityToolkit.Services/InputSimulation/ArticulatedHandPoses/ArticulatedHandPose_Flat.json

```diff
@@ -0,0 +1 @@
n":{"x":0.035030756145715716,"y":-0.025001518428325654,"z":-0.0032290546223521234},"rotation":{"x":-0.6631931662559509,"y":-0.09005288034677506,"z":-0.0027521485462784769,"w":0.7431085109710693}}},{"joint":"MiddleDistalJoint","pose":{"position":{"x":0.031546302139759067,"y":0.0013798222644254566,"z":-0.0004363078624010086},"rotation":{"x":-0.6468731164932251,"y":-0.11953263729810715,"z":-0.06937266886234284,"w":0.7504633665084839}}},{"joint":"MiddleTip","pose":{"position":{"x":0.030048875138163568,"y":0.017790958285331727,"z":0.0018172836862504483},"rotation":{"x":-0.6468731164932251,"y":-0.11953263729810715,"z":-0.06937266886234284,"w":0.7504633665084839}}},{"joint":"RingMetacarpal","pose":{"position":{"x":0.06806596368551254,"y":-0.13525664806365968,"z":-0.034837257117033008},"rotation":{"x":-0.5803540945053101,"y":0.014031633734703064,"z":0.05480925738811493,"w":0.8123965859413147}}},{"joint":"RingKnuckle","pose":{"position":{"x":0.06544187664985657,"y":-0.07453925907611847,"z":-0.013881120830774308},"rotation":{"x":-0.6466344594955444,"y":-0.03600946068763733,"z":0.02467469871044159,"w":0.7615609765052795}}},{"joint":"RingMiddleJoint","pose":{"position":{"x":0.06159381568431854,"y":-0.03093438223004341,"z":-0.006733019836246967},"rotation":{"x":-0.6550348401069641,"y":-0.06099399924278259,"z":-0.04121965169906616,"w":0.7520787715911865}}},{"joint":"RingDistalJoint","pose":{"position":{"x":0.06070023775100708,"y":-0.007464663125574589,"z":-0.003544492181390524},"rotation":{"x":-0.6712727546691895,"y":-0.05777180939912796,"z":-0.05727298930287361,"w":0.7370488047599793}}},{"joint":"RingTip","pose":{"position":{"x":0.060552775859832767,"y":0.010114867240190506,"z":-0.0019072332652285696},"rotation":{"x":-0.6712727546691895,"y":-0.05777180939912796,"z":-0.05727298930287361,"w":0.7370488047599793}}},{"joint":"PinkyMetacarpal","pose":{"position":{"x":0.07710164040327072,"y":-0.13650110363960267,"z":-0.032643478363752368},"rotation":{"x":-0.5344982147216797,"y":0.1545339822769165,"z":0.10820292681455612,"w":0.8238464593887329}}},{"joint":"PinkyKnuckle","pose":{"position":{"x":0.08530370891094208,"y":-0.08254323154687882,"z":-0.010162543505430222},"rotation":{"x":-0.6702333688735962,"y":0.05704934149980545,"z":0.006686835549771786,"w":0.7399358749389648}}},{"joint":"PinkyMiddleJoint","pose":{"position":{"x":0.08779342472553253,"y":-0.049793362617492679,"z":-0.0070251524448394779},"rotation":{"x":-0.6393072605133057,"y":0.030266048386693,"z":-0.15569603443145753,"w":0.7524937987327576}}},{"joint":"PinkyDistalJoint","pose":{"position":{"x":0.09219621121883393,"y":-0.03264733776450157,"z":-0.0037694787606596948},"rotation":{"x":-0.6555882692337036,"y":-0.0018634665757417679,"z":-0.09289215505123139,"w":0.7497090101242065}}},{"joint":"PinkyTip","pose":{"position":{"x":0.09392204880714417,"y":-0.018381092697381974,"z":-0.0017222119495272637},"rotation":{"x":-0.6555882692337036,"y":-0.0018634665757417679,"z":-0.09289215505123139,"w":0.7497090101242065}}}]}
```

Comment:
Might be good for me to clarify: are these poses a single frame, or a recording, over multiple frames? I was assuming multiple frames. For example it would be extremely useful to be able to record hands as a user is pressing a button, or for example accidentally grabs something.

I can share example code for how I did this when I was prototyping -- what I had worked for me and correctly stored timings to ensure that hand frames were replayed at exact same speed as recorded.

Reply:
Oh, this isn't the input animation recording feature yet. Here's the input animation branch:
https://github.com/lukastoenneMS/MixedRealityToolkit-Unity/tree/hand_sim_playables

I'll try to wrap the input animation branch up asap.

Reply:
A point of interest: JSON would be a bad choice for storing animation due to file size. Even with the somewhat less verbose Unity asset format the file size for hand animation is on the order of megabytes for just a few seconds of animation. This would add up quickly if we use it for test resources, so a binary format is highly desirable.